pyfr/backends/cuda/compiler.py

```diff
@@ -79,18 +81,27 @@ def compile(self, name, src, flags=[]):
 
                 raise RuntimeError(log.value.decode())
 
-            # Fetch the program size
-            ptxsz = c_size_t()
-            self.lib.nvrtcGetPTXSize(prog, ptxsz)
+            codesz = c_size_t()
+            if len([f for f in flags if 'sm_' in f]) > 0:
```

Comment:
Does it make sense to always try and get the cubin (size) first, and if that fails (returns 0) ask for the PTX?  Saves going through the flags and having to think about if later flags can supersede pervious ones.

Reply:
Yeah, defaulting to cubin and falling back to PTX makes sense to me, and should be a bit more robust. I was trying to be as minimally invasive as possible in case there was a reason to favor PTX that I wasn't aware of. I'll change it to default to cubin/fallback to PTX.

Reply:
I do not recall any specific reason why I went for PTX when I first wrote the code, so assuming it does not cause any issues lets go for requesting a cubin by default.